### PR TITLE
ISSUE-9 Introduce user actions (on create and for sending emails)

### PR DESCRIPTION
**Pull request by martijnhaex**
…. Introduce new call to send emails to User for certain user actions (call to action).

### CODE REVIEW

File: src/test/java/be/jidoka/jdk/keycloak/admin/service/KeycloakUserAdminServiceSendUserActionEmailsTest.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+import static be.jidoka.jdk.keycloak.admin.domain.CreateUserFixture.bertenBoedhoe;
+import static be.jidoka.jdk.keycloak.admin.domain.UserAction.UPDATE_PASSWORD;
+import static be.jidoka.jdk.keycloak.admin.domain.UserAction.VERIFY_EMAIL;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+class KeycloakUserAdminServiceSendUserActionEmailsTest extends IntegrationTest {
+
+	@Autowired
+	private KeycloakUserAdminService keycloakUserAdminService;
```
Comment (bastijns-jeroen) [inline src/test/java/be/jidoka/jdk/keycloak/admin/service/KeycloakUserAdminServiceSendUserActionEmailsTest.java:21]: Spin up an email container in this test. Use https://hub.docker.com/r/mailhog/mailhog for this.

File: src/main/java/be/jidoka/jdk/keycloak/admin/service/KeycloakUserAdminService.java [hunk 2]
```diff
@@ -155,4 +165,11 @@ private UserRepresentation enhanceWithClientRoles(UserRepresentation userReprese
 
 		return personalData;
 	}
+
+	private List<String> getActions(Set<UserAction> userActions) {
+		return userActions
```
[the remark this answers is not in the document]
Comment (martijnhaex) [reply src/main/java/be/jidoka/jdk/keycloak/admin/service/KeycloakUserAdminService.java:179]: Done (also for personalData (collection/map).

File: src/test/java/be/jidoka/jdk/keycloak/admin/service/KeycloakUserAdminServiceSendUserActionEmailsTest.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+	@BeforeEach
+	public void setUp() {
+		userId = keycloakUserAdminService.createUser(bertenBoedhoe());
+	}
+
+	@Test
+	void throwsExceptionWhenSMTPIsNotConfigured() {
+		SendUserActionEmailRequest sendUserActionEmailRequest = SendUserActionEmailRequestBuilder.builder()
+				.userId(userId)
+				.userActions(Set.of(UPDATE_PASSWORD, VERIFY_EMAIL))
```
[the remark this answers is not in the document]
Comment (martijnhaex) [reply src/test/java/be/jidoka/jdk/keycloak/admin/service/KeycloakUserAdminServiceSendUserActionEmailsTest.java:34]: Done.

File: src/test/java/be/jidoka/jdk/keycloak/admin/service/KeycloakUserAdminServiceSendUserActionEmailsTest.java (hunk 1 above)
Comment (martijnhaex) [reply src/test/java/be/jidoka/jdk/keycloak/admin/service/KeycloakUserAdminServiceSendUserActionEmailsTest.java:21]: Done.

**Pull request merged**